src/org/omegat/util/StringUtil.java

```diff
@@ -511,4 +511,223 @@ public static String format(String str, Object... arguments) {
+                case 0xFF85: sbfw.setCharAt(i, (char)0x30CA); break;
+                case 0xFF86: sbfw.setCharAt(i, (char)0x30CB); break;
+                case 0xFF87: sbfw.setCharAt(i, (char)0x30CC); break;
+                case 0xFF88: sbfw.setCharAt(i, (char)0x30CD); break;
+                case 0xFF89: sbfw.setCharAt(i, (char)0x30CE); break;
+                case 0xFF8A: sbfw.setCharAt(i, (char)0x30CF); break;
+                case 0xFF8B: sbfw.setCharAt(i, (char)0x30D2); break;
+                case 0xFF8C: sbfw.setCharAt(i, (char)0x30D5); break;
+                case 0xFF8D: sbfw.setCharAt(i, (char)0x30D8); break;
+                case 0xFF8E: sbfw.setCharAt(i, (char)0x30DB); break;
+                case 0xFF8F: sbfw.setCharAt(i, (char)0x30DE); break;
+                case 0xFF90: sbfw.setCharAt(i, (char)0x30DF); break;
+                case 0xFF91: sbfw.setCharAt(i, (char)0x30E0); break;
+                case 0xFF92: sbfw.setCharAt(i, (char)0x30E1); break;
+                case 0xFF93: sbfw.setCharAt(i, (char)0x30E2); break;
+                case 0xFF94: sbfw.setCharAt(i, (char)0x30E4); break;
+                case 0xFF95: sbfw.setCharAt(i, (char)0x30E6); break;
+                case 0xFF96: sbfw.setCharAt(i, (char)0x30E8); break;
+                case 0xFF97: sbfw.setCharAt(i, (char)0x30E9); break;
+                case 0xFF98: sbfw.setCharAt(i, (char)0x30EA); break;
+                case 0xFF99: sbfw.setCharAt(i, (char)0x30EB); break;
+                case 0xFF9A: sbfw.setCharAt(i, (char)0x30EC); break;
+                case 0xFF9B: sbfw.setCharAt(i, (char)0x30ED); break;
+                case 0xFF9C: sbfw.setCharAt(i, (char)0x30EF); break;
+                case 0xFF9D: sbfw.setCharAt(i, (char)0x30F3); break;
+                case 0xFF9E: sbfw.setCharAt(i, (char)0x3099); break;
+                case 0xFF9F: sbfw.setCharAt(i, (char)0x309A); break;
+                // Half width Hangul
+                case 0xFFA0: sbfw.setCharAt(i, (char)0x3164); break;
+                case 0xFFDA: sbfw.setCharAt(i, (char)0x3161); break;
+                case 0xFFDB: sbfw.setCharAt(i, (char)0x3162); break;
+                case 0xFFDC: sbfw.setCharAt(i, (char)0x3163); break;
+                // Others
+                case 0xFFE8: sbfw.setCharAt(i, (char)0x2502); break;
+                case 0xFFE9: sbfw.setCharAt(i, (char)0x2190); break;
+                case 0xFFEA: sbfw.setCharAt(i, (char)0x2191); break;
+                case 0xFFEB: sbfw.setCharAt(i, (char)0x2192); break;
+                case 0xFFEC: sbfw.setCharAt(i, (char)0x2193); break;
+                case 0xFFED: sbfw.setCharAt(i, (char)0x25A0); break;
+                case 0xFFEE: sbfw.setCharAt(i, (char)0x25CB); break;
+            }
+        }
+        // whether we have a character to be converted.
+        if ( sbhw.toString().equals(sbfw.toString()) ) {
+            return text;
+        }
+        // Create regex
+        StringBuilder sb = new StringBuilder("(");
```

Comment:
Because I think that creating full-width and half-width text in one iteration is better for performance.

Now I understand that following snippet will be better in Searcher.java
```
            // handle half-width and full-width character insensitive match
            if (expression.fullHalfWidthInsensitive) {
                if (StringUtil.containsFullWidthExpression(text)) {
                    m_matchers.add(Pattern.compile(StringUtil.getFullWidthText(text), flags).matcher(""));
                    text = StringUtil.getHalfWidthText(text);
                }
            }
            // create a matcher for the search string
            m_matchers.add(Pattern.compile(text, flags).matcher(""));
```

I can avoid creating regex using `m_matchers.add()`.
Will PR Take3 soon ...